packages/eas-json/src/utils.ts

```diff
@@ -28,23 +28,57 @@ export class EasJsonUtils {
     return resolveBuildProfile({ easJson, platform, profileName });
   }
 
-  public static getBuildProfileDeprecationWarnings(
-    buildProfile: BuildProfile,
-    profileName?: string
-  ): EasJsonDeprecationWarning[] {
+  public static async getBuildProfileDeprecationWarningsAsync(
+    easJsonAccessor: EasJsonAccessor,
+    platform: Platform,
+    profileName: string
+  ): Promise<EasJsonDeprecationWarning[]> {
     const warnings: EasJsonDeprecationWarning[] = [];
+    const buildProfile = await EasJsonUtils.getBuildProfileAsync(
+      easJsonAccessor,
+      platform,
+      profileName
+    );
+    const rawEasJson = await easJsonAccessor.readRawJsonAsync();
 
     if (buildProfile.cache?.cacheDefaultPaths !== undefined) {
       warnings.push({
         message: [
-          `The "build.${
-            profileName ?? 'production'
-          }.cache.cacheDefaultPaths" field in eas.json is deprecated and will be removed in the future.`,
+          `The "build.${profileName}.cache.cacheDefaultPaths" field in eas.json is deprecated and will be removed in the future.`,
         ],
         docsUrl: 'https://docs.expo.dev/build-reference/caching/#ios-dependencies',
       });
     }
 
+    warnings.push(...EasJsonUtils.getCustomPathsDeprecationWarnings(rawEasJson, profileName));
+
+    return warnings;
+  }
+
+  private static getCustomPathsDeprecationWarnings(
+    rawEasJson: any,
+    buildProfileName: string,
+    extendedBuildProfileName?: string
+  ): EasJsonDeprecationWarning[] {
+    const warnings = [];
+    const profileName = extendedBuildProfileName ? extendedBuildProfileName : buildProfileName;
+    if (rawEasJson.build?.[profileName]?.cache?.customPaths !== undefined) {
+      warnings.push({
+        message: [
+          `The "build.${buildProfileName}.cache.customPaths" field in eas.json is deprecated and will be removed in the future. Please use "build.${buildProfileName}.cache.paths" instead.`,
+        ],
+        docsUrl: 'https://docs.expo.dev/build-reference/eas-json/#cache',
+      });
+    }
+    if (rawEasJson.build?.[profileName]?.extends !== undefined) {
+      warnings.push(
+        ...EasJsonUtils.getCustomPathsDeprecationWarnings(
+          rawEasJson,
+          buildProfileName,
+          rawEasJson.build?.[profileName].extends
+        )
+      );
+    }
     return warnings;
   }
```

Comment:
It will not result in the same message - my version will print the profile name of the profile being checked for deprecations, your version will print the name of the profile being extended that has the deprecated fields. I guess it's debatable which name is more useful to the user, but I would say the name of the final (extending) profile, not the base (extended) one

Reply:
Well, in that case I can make the change

Reply:
Updated